### PR TITLE
Add synchronization tools

### PR DESCRIPTION
**Pull request by TrySound**
1. New method store.createTransactionFromChanges to apply changes created remotely
2. Additional "source" parameter to identify in subscribe callback the source of changes and for example prevent infinite loop for two-way patches.
3. Provide revise patches in changes to support undo of remote changes

### CODE REVIEW

File: src/store.ts [hunk 1]
```diff
@@ -22,6 +22,7 @@ type UnwrapContainers<Containers extends Array<ValueContainer<unknown>>> = {
 
 export class Store {
   registry: Map<ValueContainer<Any>, string>;
+  containers: Map<string, ValueContainer<Any>>;
```
Comment (kof) [inline src/store.ts]: naming got confusing between these two

Comment (kof) [reply src/store.ts]: seems registry is containerNamesByRef and containers is caontainerRefsByName or something but  maybe we can simplify both and use one registry format?

Comment (TrySound) [reply src/store.ts]: Or rename registry to namespaces?

Comment (kof) [reply src/store.ts]: yeah, sounds good

**Review by kof — APPROVED** [review]

**Pull request merged**